src/GovUk.Education.ExploreEducationStatistics.Data.Importer/Utils/DbUtils.cs

```diff
@@ -24,5 +26,35 @@ public static ContentDbContext CreateContentDbContext()
 
             return new ContentDbContext(optionsBuilder.Options);
         }
+
+        /**
+         * Obtains an exclusive lock within a new transaction and stops other transactions needing to acquire the
+         * same lock to run until they can obtain the lock.  The lock is released upon a transaction being committed
+         * or rolled back.
+         */
+        public static Task<TResult> ExecuteWithExclusiveLock<TDbContext, TResult>(
+            TDbContext dbContext,
+            string lockName,
+            Func<TDbContext, Task<TResult>> action) 
+            where TDbContext : DbContext
+        {
+            return dbContext.Database.CreateExecutionStrategy().ExecuteAsync(async () =>
+            {
+                await using var transaction = await dbContext.Database.BeginTransactionAsync();
+                await dbContext.Database.ExecuteSqlRawAsync($"exec sp_getapplock '{lockName}', 'exclusive'");
```

Comment:
It's pretty cool SQL Server lets you write locks like this! :exploding_head:

Reply:
Yeah - with great power comes great responsibility!